src/test/java/spoon/test/refactoring/RenameLocalVariableRefactorTest.java

```diff
@@ -0,0 +1,255 @@
+package spoon.test.refactoring;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+import org.junit.Test;
+
+import spoon.Launcher;
+import spoon.OutputType;
+import spoon.SpoonException;
+import spoon.SpoonModelBuilder;
+import spoon.refactoring.RenameLocalVariableRefactor;
+import spoon.reflect.code.CtLocalVariable;
+import spoon.reflect.declaration.CtAnnotation;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtVariable;
+import spoon.reflect.reference.CtTypeReference;
+import spoon.test.refactoring.testclasses.TestTryRename;
+import spoon.test.refactoring.testclasses.RenameLocalVariableRefactorTestSubject;
+import spoon.testing.utils.ModelUtils;
+
+public class RenameLocalVariableRefactorTest
+{
+	@Test
+	public void testModelConsistency() throws Throwable {
+		//contract: check that all assertions in all methods of the RenameLocalVariableRefactorTestSubject are correct
+		new RenameLocalVariableRefactorTestSubject().checkModelConsistency();
+	}
+	
+	/**
+	 * If you need to debug behavior of refactoring on the exact method and variable in the {@link RenameLocalVariableRefactorTestSubject} model,
+	 * then provide
+	 * 1) name of method of {@link RenameLocalVariableRefactorTestSubject}
+	 * 2) original name variable in the method
+	 * 3) new name of variable in the method
+	 * then put breakpoint on the line `this.getClass();` below and the debugger stops just before 
+	 * the to be inspected refactoring starts
+	 */
+	private String[] DEBUG = new String[]{/*"nestedClassMethodWithoutRefs", "var3", "var1"*/};
+
+	/**
+	 * The {@link RenameLocalVariableRefactorTestSubject} class is loaded as spoon model. Then:
+	 * - It looks for each CtVariable and it's CtAnnotation and tries to rename that variable to the name defined by annotation.
+	 * - If the annotation name is prefixed with "-", then that refactoring should fail.
+	 * - If the annotation name is not prefixed, then that refactoring should pass.
+	 * If it behaves different then expected, then this test fails
+	 */
+	@Test
+	public void testRenameAllLocalVariablesOfRenameTestSubject() throws Exception {

```

Comment:
by the way, it's in essence a parametrized test, where RenameLocalVariableRefactorTestSubject contains the test parameters. Really clever. It would be even more explicit and maintainable written as  [`@Parametrized`](https://github.com/junit-team/junit4/wiki/parameterized-tests).